File: blog-parent/blog-admin-web/src/main/webapp/js/service/blogService.js
//服务层
app.service('blogService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../blog/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../blog/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../blog/findOne.do?id='+id);
	}
	//查询实体
	this.findOneByAccount=function(account){
		return $http.get('../blog/findOne.do?account='+account);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../blog/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../blog/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../blog/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../blog/search.do?page='+page+"&rows="+rows, searchEntity);
	}    
	
});

File: blog-parent/blog-interface/src/main/java/com/blog/service/BlogService.java
package com.blog.service;

import java.util.List;

import com.blog.pojo.Blog;

import entity.PageResult;

public interface BlogService {
	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Blog> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Blog blog);
	
	
	/**
	 * 修改
	 */
	public void update(Blog blog);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Blog findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Blog blog, int pageNum,int pageSize);
}
